modules/silicon_design/main.tf

```diff
@@ -182,6 +200,37 @@ resource "google_project_iam_member" "module_role2" {
   role     = "roles/viewer"
 }
 
+resource "google_project_service_identity" "sa_cloudbuild_identity" {
+  provider = google-beta
+  project  = local.project.project_id  
+  service = "cloudbuild.googleapis.com"
+}
+
+resource "google_project_iam_member" "sa_cloudbuild_permissions" {
+  for_each = toset(local.cloudbuild_sa_project_roles)
+  project  = local.project.project_id
+  member   =  "serviceAccount:${google_project_service_identity.sa_cloudbuild_identity.email}"
+  role     = each.value
+}
+
+resource "google_service_account_iam_member" "sa_cloudbuild_image_builder_access" {
+  member             = "serviceAccount:${google_project_service_identity.sa_cloudbuild_identity.email}"
+  role               = "roles/iam.serviceAccountUser"
+  service_account_id = google_service_account.sa_image_builder_identity.id
+}
+
+resource "google_service_account" "sa_image_builder_identity" {
+  project    = local.project.project_id
+  account_id = "sa-image-builder-identity"
+}
+
+resource "google_project_iam_member" "sa_image_builder_permissions" {
+  for_each = toset(local.image_builder_sa_project_roles)
+  project  = local.project.project_id
+  member   = "serviceAccount:${google_service_account.sa_image_builder_identity.email}"
+  role     = each.value
+}
+
 resource "google_notebooks_instance" "ai_notebook" {
```

Comment:
Is there a reason why creation of `google_notebooks_instance.ai_notebook` resource is depended on  `null_resource.build_and_push_image` ?

Reply:
Because we need the container image to be built (thru the cloud build trigger by the null_resource) in order to be able to start the notebook instance.

Reply:
Sounds good! that make sense..